core/templates/services/context.service.spec.ts

```diff
@@ -246,12 +260,47 @@ describe('Context service', () => {
       });
   });
 
+  describe('behavior in different pages', () => {
+    beforeEach(() => {
+      ecs = TestBed.get(ContextService);
+      urlService = TestBed.get(UrlService);
+    });
+
+    it('should correctly retrieve the page context', () => {
+      spyOn(urlService, 'getPathname').and.returnValue('/question_editor/123');
```

Comment:
Here and below, should we have a "precheck". e.g. we check that the context is not X and then spy on urlService and finally check that it is X?

Reply:
Also here, Can you please elaborate on pre-check over here?

Reply:
Done!